imports/ui/components/study_plan/Tab.jsx

```diff
@@ -189,39 +188,71 @@ export default class Tab extends React.Component {
    */
   renderDropDownMenuSelection(className, iconClassName, iconStyle, text, clickHandler) {
     return (
-      <div className={className} onClick={clickHandler.bind(this)}>
-        <i className={iconClassName} style={iconStyle}></i>
-        <div>{text}</div>
-      </div>
+        <div className={className} onClick={clickHandler.bind(this)}>
+          <i className={iconClassName} style={iconStyle}></i>
+          <div>{text}</div>
+        </div>
     )
   }
 
   render() {
+    var onRightClick = this.props.onRightClickTab;
     return (
       /* ReactClickOut listens for and handles all click-outside events */
-      <ReactClickOut onClickOut={this.handleOnBlurDropdown.bind(this)} >
-        <li className='nav-item' style={this.props.tabStyle}
-            onClick={this.props.onClickTab}
-            onMouseEnter={this.handleOnMouseEnter.bind(this)}
-            onMouseLeave={this.handleOnMouseLeave.bind(this)}>
-          <a className={this.setActiveTabClass("nav-link")} role='tab'>
-            <span className={this.props.navSpanClass} style={this.props.navSpanStyle}>
-
-              {/* The label of this Tab is rendered here */}
-              { this.props.tabTitle }
-
-              {/* Render dropdown caret */}
-              { this.props.enabledDropdown && this.props.enabledMouseOver &&
-                !this.props.isEditingPlanName ?
-                this.renderDropdownCaret() : null }
-
-              {/* Toggle Dropdown menu for this tab */}
-              { this.props.enabledDropdown && this.state.onClickDropdown ?
-                this.renderDropDownMenu() : null }
-
-            </span>
-          </a>
-        </li>
+      <ReactClickOut onClickOut={this.handleOnBlurDropdown.bind(this)}>
+          <li className='nav-item'
+              onClick={this.props.onClickTab}
+              onContextMenu={onRightClick ? onRightClick : this.handleRightClick.bind(this)}
+              onMouseEnter={this.handleOnMouseEnter.bind(this)}
+              onMouseLeave={this.handleOnMouseLeave.bind(this)}>
+            <a className={this.setActiveTabClass('nav-link')} role='tab'>
+              <span className={this.props.navSpanClass} style={this.props.navSpanStyle}
+                  ref={(el)=> {
```

Comment:
Might need to comment about what's going on here/why there is an extra computation needed for the tab dimensions! Maybe even abstract this part into a helper function above with some code documentation. 

I'm guessing that it has to do with the position and width of the Tab. Small issue and we can discuss again tmr!